launch/launch/logging/__init__.py

```diff
@@ -331,7 +333,12 @@ def get_screen_handler():
     See launch_config() documentation for screen logging configuration.
     """
     if launch_config.screen_handler is None:
-        launch_config.screen_handler = handlers.StreamHandler(sys.stdout)
+        stream = codecs.StreamWriter(sys.stdout, errors='replace')
+        stream.encode = lambda msg, errors='replace': (
+            msg.encode(locale.getpreferredencoding(False), errors).decode(
+                errors=errors),
+            msg)
+        launch_config.screen_handler = handlers.StreamHandler(stream)
```

Comment:
This building a custom `StreamWriter` seems like a departure from normal use, is there a SO or example somewhere that you're pulling from?

Reply:
Since the `StreamHandler` itself doesn't support an encoding the stream has to apply the encoding: see https://www.reddit.com/r/learnpython/comments/95uzy2/logging_how_to_define_encoding_of/